data-prepper-plugins/log-generator-source/src/test/java/com/amazon/dataprepper/plugins/source/loggenerator/LogGeneratorSourceTest.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.amazon.dataprepper.plugins.source.loggenerator;
+
+
+import com.amazon.dataprepper.model.event.JacksonEvent;
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.configuration.PluginModel;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.plugin.PluginFactory;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.plugins.buffer.blockingbuffer.BlockingBuffer;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.TimeoutException;
+import java.time.Duration;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.lenient;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.junit.Assert.assertEquals;
+
+public class LogGeneratorSourceTest {
+    private LogGeneratorSourceConfig sourceConfig;
+    private PluginMetrics pluginMetrics;
+    private PluginFactory pluginFactory;
+    private LogGeneratorSource logGeneratorSource;
+
+    @BeforeEach
+    public void setup() {
+        sourceConfig = mock(LogGeneratorSourceConfig.class);
+        pluginMetrics = mock(PluginMetrics.class);
+        pluginFactory = mock(PluginFactory.class);
+
+        PluginModel mockLogPluginModel = mock(PluginModel.class);
+        when(sourceConfig.getLogType()).thenReturn(mockLogPluginModel);
+
+        when(mockLogPluginModel.getPluginName()).thenReturn(
+                "TestPluginName"
+        );
+        Map<String, Object> emptyMap = new HashMap<>();
+        when(mockLogPluginModel.getPluginSettings()).thenReturn(
+                emptyMap
+        );
+        LogTypeGenerator commonApacheLogTypeGenerator = mock(LogTypeGenerator.class);
+
+
+        when(pluginFactory.loadPlugin(any(), any(PluginSetting.class))).thenReturn(commonApacheLogTypeGenerator);
+        String hardcodedEvent = "127.0.0.1 user pwd [date] \"GET /list HTTP/1.1\" 404 4043";
+        Event stubbedJackson = JacksonEvent.fromMessage(hardcodedEvent);
+        lenient().when(commonApacheLogTypeGenerator.generateEvent()).thenReturn(stubbedJackson); // JacksonEvent type
+
+        logGeneratorSource = new LogGeneratorSource(sourceConfig, pluginMetrics, pluginFactory);
```

Comment:
Nit: In general the `createObjectUnderTest` pattern is preferred for unit tests. Here is an example of this pattern in use (https://github.com/opensearch-project/data-prepper/blob/385cf6d31d70b6d5ebf9fde8b5c2117b113bf961/data-prepper-plugins/aggregate-processor/src/test/java/com/amazon/dataprepper/plugins/processor/aggregate/AggregateProcessorTest.java#L136)